snapcraft/internal/errors.py

```diff
@@ -45,6 +47,57 @@ def __init__(self, step):
         super().__init__(step=step)
 
 
+class StepOutdatedError(SnapcraftError):
+
+    fmt = (
+        'The {step!r} step of {part!r} is out of date:\n'
+        '{report}'
+        'In order to continue, please clean that part\'s {step!r} step '
+        'by running:\n'
+        'snapcraft clean {parts_names} -s {step}\n'
+    )
+
+    def __init__(self, step, part,
```

Comment:
please add an `*` after self to force all of these to be kwargs (API clarity in the long wrong)

Reply:
Hmm I had it in there before but it didn't seem to work... I guess it was unrelated. Added the * back!